public/libs/future.js

```diff
@@ -20,6 +20,19 @@ class Future extends Promise {
         this._reject = reject;
     }
 
+    static withMinDuration(promise, minDuration) {
```

Comment:
Maybe add a short docstring with a note/warning that this does not run inside the AngularJS loop.